WordPress/gutenberg
# Post template: Adjust the editor preview on taxonomy archive templates

## What?
This pull request adds two conditions to the edit.js file in the post template block, to ensure that the
preview in the editor shows the correct posts on tag and post format templates.

Closes https://github.com/WordPress/gutenberg/issues/65320

## Why?
Archive templates for tags and post formats showed the wrong posts in the editor when the query loop was set to inherit (Query type: Default). Instead of listing the posts that are assigned the tag or format, it showed the latest posts.

## How?

If a the template slug starts with `tag-`, the id of the tag is fetched using `getEntityRecords`. The ID is then added to the query, and the correct posts are returned and shown in the preview.
This is identical to the current solution for the category templates.

If a template slug starts with `taxonomy-post_format`, the format is collected from the template slug and added to the query.

## Testing Instructions
Activate a block theme.
Create a new post with a tag.
Go to Appearance > Editor > Templates.
Select "Add new template", then "Tag Archives" and "for a specific item".
Create the new template for the tag you added to your post.

Now depending on your theme you should be able to select a pattern for an archive, 
this pattern will include query loops and post template blocks.
If not, you will need to add these blocks.
Just to be sure, double check that the query loop is set to "Query type: Default". Not custom.
Confirm that the preview in the editor shows post(s) with the correct tag.

To test the post format archive, you need a theme that has supports for post formats.
If your theme is missing this feature, register post formats by adding this example code to the theme's functions.php file.

```
function twentytwentyfour_setup(){
	add_theme_support( 'post-formats', array( 'standard', 'aside', 'gallery', 'audio', 'video', 'link', 'image', 'chat', 'status', 'quote' ) );
}

add_action( 'after_setup_theme', 'twentytwentyfour_setup' );
```
Set the permalinks on the test installation to Post name.

Create a few posts and assign them the link post format.

Next, add a link post format archive template inside the theme's templates folder.
templates/taxonomy-post_format-post-format-link.html

Go to Appearance > Editor > Templates, locate and edit the template.
Add a query loop block. In the block settings sidebar, make sure that "Query type: Default" is enabled.
Check if the query loop shows the correct posts, it should only show the posts that you assigned the link format to.
Save.
